far/filelist.cpp

```diff
@@ -1953,7 +1953,7 @@ bool FileList::ProcessKey(const Manager::Key& Key)
 						if (PluginMode && UploadFile)
 						{
 							PluginPanelItemHolderHeap PanelItem;
-							const auto strSaveDir = os::fs::get_current_directory();
+							const auto& strSaveDir = os::fs::get_current_directory();
```

Comment:
This [doesn't really change anything](https://godbolt.org/z/KhTr7qGr5).

Reply:
Hmm... If I am not mistaken, `os::fs::get_current_directory` is [declared](https://github.com/FarGroup/FarManager/blob/33df9f5537981b4b3e33bbaf7d7c01772aec2475/far/platform.fs.hpp#L496) as
```
const string& get_current_directory();
```
If so, declaring reference here [makes whole lot of difference](https://godbolt.org/z/eqb4WbTed).

Reply:
@MKadaner,
I agree with you, my IDE when hovering over `const auto` returns that the type will be `const string` without reference.
Tested on Qt Creator 7.0.0 and VS 2019

Reply:
@MKadaner, yes, you're right.

However, then it's another argument against changing this: as the name of the variable suggests, we save the "current directory" value here, apparently because the code below sets it to something else, and afterwards we restore it from the saved copy. If we change it to a reference, it will no longer "save" anything and the logic will be broken.

Reply:
@alabuzhev Oh! That's for sure. My bad; did not look further down. :(